packages/engine/src/camera/systems/CameraSystem.ts

```diff
@@ -51,6 +51,17 @@ const followCameraBehavior = (entity: Entity) => {
     FollowCameraComponent
   ) as FollowCameraComponent
 
+  if (CameraSystem.instance.updateCameraMode) {
+    let index = 0
+    for (const key of Object.keys(CameraModes)) {
+      if (index === CameraSystem.instance.cameraModeIndex && CameraSystem.instance.updateCameraMode) {
```

Comment:
Why are we checking `CameraSystem.instance.updateCameraMode` here? We have already checked on the if condition above.